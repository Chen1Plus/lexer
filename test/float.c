float a = 123.456;
float b = 0.456;
float c = .456;
float d = 456.;
float e = 0.;
float f = .0;
float g =.; // invalid case

float a = -123.456;
float b = -0.456;
float c = -.456;
float d = -456.;
float e = -0.;
float f = -.0;

// invalid case
float g = -.;
float h = - .00;
float i = - 0.00;
float j = - 0.;
