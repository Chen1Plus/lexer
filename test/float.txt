123.456
0.456
.456
456.
0.
.0

+123.456
+0.456
+.456
+456.
+0.
+.0

-123.456
-0.456
-.456
-456.
-0.
-.0
