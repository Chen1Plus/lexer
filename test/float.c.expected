#float:123.456
#float:0.456
#float:.456
#float:456.
#float:0.
#float:.0
#float:-123.456
#float:-0.456
#float:-.456
#float:-456.
#float:-0.
#float:-.0
#float:.00
#float:0.00
#float:0.
